### PR TITLE
[hps] Add new port

### LINKED ISSUE #23346: [New Port Request] hps
Library name: hps

Library description: A C++11 High Performance Serialization Library

Source repository URL: https://github.com/jl2922/hps

Project homepage (if different from the source repository):

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by PrateekJoshi**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #23346

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  linux , windows. No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: versions/h-/hps.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "versions": [
+    {
+      "git-tree": "828d03d4dca08052797af248bd22e64dfc2f76c8",
+      "version-date": "2022-01-18",
+      "port-version": 0
+    },
+    {
+      "git-tree": "8cbb036bddf59db82a3b0035dd872901fe7ade3e",
+      "version": "1.0.0",
+      "port-version": 0
```
Comment (Thomas1664) [inline versions/h-/hps.json:11]:
```suggestion
```

**Review by Thomas1664 — COMMENTED** [review]
Can you please add quotes around each path that starts with `${CURRENT_PACKAGES_DIR}` or `${SOURCE_PATH}`?

**Comment by PrateekJoshi** [comment]
> Can you please add quotes around each path that starts with `${CURRENT_PACKAGES_DIR}` or `${SOURCE_PATH}`?

@Thomas1664 Done

**Review by Thomas1664 — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**